src/main/java/no/digipost/signature/client/core/internal/SignatureHttpClient.java

```diff
@@ -43,14 +47,25 @@ public static Client create(ClientConfiguration config) {
                     .sslContext(sslcontext)
                     .hostnameVerifier(NoopHostnameVerifier.INSTANCE)
                     .build();
-        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException | UnrecoverableKeyException e) {
-            throw new RuntimeException(e.getClass().getSimpleName() + ": " + e.getMessage(), e);
+        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException | UnrecoverableKeyException e) {
+            if (e instanceof UnrecoverableKeyException && "Given final block not properly padded".equals(e.getMessage())) {
+                throw new KeyException("Unable to load key from keystore. Possible causes:\n" +
+                        "* Wrong password for private key (the password for the keystore and the private key may not be the same)\n" +
+                        "* Multiple private keys in the keystore with different passwords (private keys in the same key store must have the same password)", e);
+            } else {
+                throw new KeyException("Unable to load key from keystore", e);
```

Comment:
I usually just concat `e.getClass().getSimpleName() + ": " + e.getMessage()` to the end of my own message, to get both the context (your own message), and the reason (from the wrapped exception) in the top-level exception message. Something like this:
```
throw new KeyException("Unable to load key from keystore, because " + e.getClass().getSimpleName() + ": " + e.getMessage(), e);
```
I typically do this for "if all else fails" exception handling, so I'd just leave the special case handling of the UnrecoverableKeyException as-is.